merlin/models/tf/models/base.py

```diff
@@ -1397,6 +1399,146 @@ def to_top_k_recommender(
         return recommender
 
 
+@tf.keras.utils.register_keras_serializable(package="merlin_models")
+class RetrievalModelV2(Model):
+    def __init__(
+        self,
+        *,
+        query: Union[Encoder, tf.keras.layers.Layer],
+        output: Union[ModelOutput, tf.keras.layers.Layer],
+        candidate: Optional[Union[Encoder, tf.keras.layers.Layer]] = None,
+        query_name="query",
+        candidate_name="candidate",
+        pre: Optional[tf.keras.layers.Layer] = None,
+        post: Optional[tf.keras.layers.Layer] = None,
+        **kwargs,
+    ):
+        if isinstance(output, ContrastiveOutput):
+            query_name = output.query_name
+            candidate_name = output.candidate_name
+
+        if query and candidate:
+            encoder = ParallelBlock({query_name: query, candidate_name: candidate})
```

Comment:
If I'm understanding this correctly, does this need to be a ParallelBlock, or could it be a dictionary? 

If they don't need to be paired together in that way it might be clearer to follow the the implementation if these were assigned to different properties (`self._query_encoder`, `self._candidate_encoder`)

Reply:
It needs to be a Block, but we could implement a custom block. Something lilke:

```python
class BiEncoder(ParallelBlock):
  …

  @property
  def query_encoder(self):
    return self[“query”]

  @property
  def candidate_encoder(self):
    return self[“candidate"]
```


Do you think that’s cleaner?

Reply:
That may help clarify layer names if you were to look at the model representation in that way. 

In this case, I missed the connection to the `__init__` method below where this is the first block of the model.